test-packages/type-tests/test/v2/batch.test-d.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  changeset as otherServiceChangeset,
+  multipleSchemasService
+} from '@sap-cloud-sdk/test-services-odata-v2/multiple-schemas-service';
+import {
+  batch,
+  changeset as testEntityChangeset,
+  TestEntity,
+  testService
+} from '@sap-cloud-sdk/test-services-odata-v2/test-service';
+import {
+  mergeDefaultDeSerializersWith,
+  ReadResponse,
+  WriteResponses
+} from '@sap-cloud-sdk/odata-v2/internal';
+import { expectType } from 'tsd';
+import { BatchChangeSet } from '@sap-cloud-sdk/odata-common';
+import {
+  BatchResponse,
+  DefaultDeSerializers,
+  ODataBatchRequestBuilder
+} from '@sap-cloud-sdk/odata-v2';
+import {
+  CustomDeSerializerV2,
+  DefaultDeSerializerV2
+} from '../duplicated-types';
+
+const { testEntityApi } = testService();
+const { multiSchemaTestEntityApi } = multipleSchemasService();
+
+export const customTestDeSerializers: CustomDeSerializerV2 =
+  mergeDefaultDeSerializersWith({
+    'Edm.String': {
+      deserialize: () => 100,
+      serialize: (value: number) => value.toString(),
+      serializeToUri: (value: number, serialize) => `'URI(${serialize(value)})'`
+    }
+  });
+
+const createTestEntity = testEntityApi
+  .requestBuilder()
+  .create(testEntityApi.entityBuilder().build());
+const createTestEntityFromOtherService = multiSchemaTestEntityApi
+  .requestBuilder()
+  .create(multiSchemaTestEntityApi.entityBuilder().build());
+
+const changeSetTestEntity = testEntityChangeset(createTestEntity);
+expectType<BatchChangeSet<DefaultDeSerializerV2>>(changeSetTestEntity);
+
+expectType<BatchChangeSet<DefaultDeSerializerV2>>(
+  testEntityChangeset(createTestEntity, createTestEntity)
+);
+
+expectType<BatchChangeSet<DefaultDeSerializerV2>>(
+  testEntityChangeset([createTestEntity, createTestEntity])
+);
+
+expectType<BatchChangeSet<DefaultDeSerializerV2>>(
+  otherServiceChangeset(createTestEntityFromOtherService)
+);
+
+// // $ExpectError
```

Comment:
[req] Either remove these or convert them to a TODO, otherwise will just be irrelevant for everyone